sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,1739 @@
+        destination.write_i64::<BigEndian>(self.created)?;
+        destination.write_i64::<BigEndian>(self.modified)?;
+        destination.write_u16::<BigEndian>(self.unitsPerEm)?;
+        destination.write_i16::<BigEndian>(self.xMin)?;
+        destination.write_i16::<BigEndian>(self.yMin)?;
+        destination.write_i16::<BigEndian>(self.xMax)?;
+        destination.write_i16::<BigEndian>(self.yMax)?;
+        destination.write_u16::<BigEndian>(self.macStyle)?;
+        destination.write_u16::<BigEndian>(self.lowestRecPPEM)?;
+        destination.write_i16::<BigEndian>(self.fontDirectionHint)?;
+        destination.write_i16::<BigEndian>(self.indexToLocFormat)?;
+        destination.write_i16::<BigEndian>(self.glyphDataFormat)?;
+        Ok(())
+    }
+}
+
+/// Generic font table with unknown contents.
+#[derive(Debug)]
+struct TableUnspecified {
+    data: Vec<u8>,
+}
+
+/// Any font table.
+impl TableUnspecified {
+    /// Creates an unspecified table from the given stream.
+    pub fn _new_from_reader<T: Read + Seek + ?Sized>(_source_stream: &mut T) -> core::result::Result<TableUnspecified, Error> {
+        Err(Error::FontLoadError)?
+    }
+    /// Wrnite
+    fn write<TDest: Write + ?Sized>(&mut self, destination: &mut TDest) -> Result<()> {
+        Ok(destination.write_all(&self.data[..])?)
+    }
+}
+
+/// Possible tables
+#[derive(Debug)]
+enum Table {
+    /// 'C2PA' table
+    C2PA(TableC2PA),
+    /// 'head' table
+    //Head(TableHead),
+    /// any other table
+    Unspecified(TableUnspecified),
+    /// WOFFHeader - not really a table
+    WoffHeader(WoffHeader),
+}
+
+/// Font abstraction sufficient for SFNT (TrueType/OpenType), WOFF (1 and, with
+/// enhancements, perhaps 2) and EOT fonts. Potentially composable to support
+/// TrueType/OpenType Collections.
+/// 
+/// TBD - Font should be a trait, not a struct, and should then be implemented
+/// by struct Woff, struct Sfnt, struct Woff2, etc.
+struct Font {
+    /// Magic number for this font's container format
+    magic: Magic,
+    /// All the Tables in this font, keyed by TableTag.
+    // TBD - WOFF2 - For this format, the Table Directory entries are not
+    // sorted by tag; rather, their order determines the physical order of the
+    // tables themselves. Therefore, a BTreeMap keyed by tag is not appropriate;
+    // we need to generalize the concept of table order across font container
+    // types. Design considerations include:
+    // - Any motivation to _not_ ensure that SFNTs/WOFF1s are sorted correctly;
+    //   if our goal is simply to describe the provenance _of an already-existing asset_,
+    //   then perhaps we need to permit incorrectly-ordered SFNTs/WOFF1s? In that
+    //   case, then perhaps the selection of BTreeMap versus Vec is done at
+    //   construction time, ensuring the desired behavior?
+    // - Otherwise, we could just use BTreeMap for SFNT/WOFF1 and Vec for WOFF2
+    // - Other matters?
+    tables: BTreeMap<TableTag, Table>,
```

Comment:
Yeah, I imagine as we go along, we will just end up using the order given, so we are not modifying the font too much.

Reply:
Well, my thinking is that when we finally make `Font` into a `trait`, with an underlying SFNT/WOFF/whatever engine, then the engine-layer will choose how to support the `Font` abstraction's key/value table storage. (You could then have "stable SFNT" and "fixup SFNT" engines that differed very little...)